etest/profile.py

```diff
@@ -0,0 +1,124 @@
+"""Profile definitions and builders."""
+
+import logging
+from enum import Enum
+
+
+class _arm_mappings(Enum):
+    armv5 = "armv5tel"
+    armv6 = "armv6j_hardfp"
+    armv7 = "armv7a_hardfp"
+    arm64 = "arm64"
+
+
+class InvalidProfileError(ValueError):
+    """The profile is invalid."""
+
+    pass
+
+
+class Profile:
+    """A class to generate a profile string and information.
+
+    Uses cli options.
+    """
+
+    def __init__(
+        self,
+        logger: logging.Logger,
+        strict: bool,
+        arch: str,
+        libc: str,
+        hardened: bool,
+        multilib: bool,
+        systemd: bool,
+    ) -> None:
+        """Initialize parameters."""
+        self.logger = logger
```

Comment:
Why do we carry a logger on an object?  That doesn't make any sense.  The logger should be attached to the module and then leveraged that way to control logging.